### PR TITLE
Fix error when attempting to read elided onnx constants

### PR DESCRIPTION
**Pull request by renxida**

### CODE REVIEW

**Comment by renxida** [comment]
Fixes https://github.com/llvm/torch-mlir/issues/3394

**Comment by renxida** [comment]
Test:

Source onnx:
```mlir
func.func @test_func() -> !torch.vtensor<[256],f32> attributes {torch.onnx_meta.ir_version = 8 : si64, torch.onnx_meta.opset_version = 17 : si64, torch.onnx_meta.producer_name = "pytorch", torch.onnx_meta.producer_version = "1.13.1"} {
    %9 = torch.operator "onnx.Constant"() {torch.onnx.value = dense_resource<__9> : tensor<256xf32>} : () -> !torch.vtensor<[256],f32> 
    %10 = torch.operator "onnx.Constant"() {torch.onnx.value = dense_resource<__19> : tensor<256xf32>} : () -> !torch.vtensor<[256],f32> 
    %11 = torch.operator "onnx.Add"(%9, %10) : (!torch.vtensor<[256],f32>, !torch.vtensor<[256],f32>) -> !torch.vtensor<[256],f32>
    return %11 : !torch.vtensor<[256],f32>
}

{-#
  dialect_resources: {
    builtin: {
        
    }
  }
#-}


```

Result torch-mlir:
```
module {
  func.func @test_func() -> !torch.vtensor<[256],f32> attributes {torch.onnx_meta.ir_version = 8 : si64, torch.onnx_meta.opset_version = 17 : si64, torch.onnx_meta.producer_name = "pytorch", torch.onnx_meta.producer_version = "1.13.1"} {
    %0 = torch.vtensor.literal(dense_resource<__onnx_constant_not_found_and_may_be_elided__> : tensor<256xf32>) : !torch.vtensor<[256],f32>
    %1 = torch.vtensor.literal(dense_resource<__onnx_constant_not_found_and_may_be_elided___1> : tensor<256xf32>) : !torch.vtensor<[256],f32>
    %int1 = torch.constant.int 1
    %2 = torch.aten.add.Tensor %0, %1, %int1 : !torch.vtensor<[256],f32>, !torch.vtensor<[256],f32>, !torch.int -> !torch.vtensor<[256],f32>
    return %2 : !torch.vtensor<[256],f32>
  }
}

{-#
  dialect_resources: {
    builtin: {
      __onnx_constant_not_found_and_may_be_elided__: "0x00000000",
      __onnx_constant_not_found_and_may_be_elided___1: "0x00000000"
    }
  }
#-}
```

**Comment by renxida** [comment]
I don't like that it adds 
```
{-#
  dialect_resources: {
    builtin: {
      __onnx_constant_not_found_and_may_be_elided__: "0x00000000",
      __onnx_constant_not_found_and_may_be_elided___1: "0x00000000"
    }
  }
#-}
```
and feels hacky but at least it works.

File: lib/Conversion/TorchToArith/TorchToArith.cpp [hunk 1]
```diff
@@ -25,6 +25,9 @@
 #include "torch-mlir/Dialect/TorchConversion/IR/TorchConversionDialect.h"
 #include "torch-mlir/Dialect/TorchConversion/Transforms/BackendTypeConversion.h"
 
+// for llvm::dbgs
+#include "llvm/Support/Debug.h"
```
Comment (zjgarvey) [inline lib/Conversion/TorchToArith/TorchToArith.cpp:29]: This looks like a debugging artifact

**Review by zjgarvey — COMMENTED** [review]
Nice, I'm glad you got something to work! If anything, this is still massively better than a crash. 

Do the empty dense attr resources get cleaned up if you run further `--mlir-elide-...` passes?

**Comment by renxida** [comment]
> Do the empty dense attr resources get cleaned up if you run further `--mlir-elide-...` passes?
Yup!  Just tested & it does:

image

**Review by zjgarvey — APPROVED** [review]
I tried it out and it does what I want it to do. 

Thanks for the QoL improvement, Xida.

**Pull request merged**